deps/rabbit/src/rabbit_peer_discovery.erl

```diff
@@ -423,27 +425,45 @@ query_node_props(Nodes) when Nodes =/= [] ->
 query_node_props([]) ->
     [].
 
-maybe_add_tls_arguments(VMArgs0) ->
+maybe_add_proto_dist_arguments(VMArgs) ->
     case init:get_argument(proto_dist) of
-        {ok, [["inet_tls"]]} ->
-            add_tls_arguments(inet_tls, VMArgs0);
-        {ok, [["inet6_tls"]]} ->
-            add_tls_arguments(inet6_tls, VMArgs0);
+        {ok, [[Val]]} ->
+            %% Note:
+            %%
+            %% We can't use list_to_existing_atom/1 because
+            %% the atom doesn't actually already exist.
+            %%
+            %% See net_kernel.erl / protocol_childspecs/1
+            Mod = list_to_atom(Val ++ "_dist"),
+            ModDir = filename:dirname(code:which(Mod)),
```

Comment:
@lukebakken: If the atom may not exist, is the module loaded then? I don't think we can compare the context of `net_kernel` to the Peer discovery subsystem. `net_kernel` is used very early and is responsible for loading the distribution module and setting the distribution up.

When peer discovery runs, the distribution module must be loaded already. Thus I'm surprised that the atom doesn't exist.

I mean, it's not a big deal, we can use `list_to_atom/1` instead of `list_to_existing_atom/1`, but the comment seems incorrect in the context of peer discovery (it can be true in `net_kernel`).

Where you able to reproduce an exception with `list_to_existing_atom/1`? I understand there was a mistake in the original code where the `_dist` suffix was not appended.

Reply:
> Where you able to reproduce an exception with list_to_existing_atom/1

Aha, I bet I got the exception because the `_dist` suffix had not been added. I will re-test with `list_to_existing_atom/1`

Reply:
Yep, that was it!